library/choose_disk.py

```diff
@@ -0,0 +1,651 @@
+            final_disks[current_index] = physical_disks[physical_disk]
+
+    return final_disks
+
+
+def fake_device(legacy_devices, ceph_type):
+    '''
+    In case of legacy block device names, let's create an internal faked
+    entry with a 'bdev' entry filled with the actual path. This will be used to
+    make a match later on.
+    '''
+    devices = {}
+    count = 0
+    for device in legacy_devices:
+        devices["%s_%d" % (ceph_type, count)] = {"bdev": os.path.dirname(device)+"/"+os.path.basename(device)}
+        count = count + 1
+
+    return devices
+
+
+def show_resulting_devices(matched_devices, physical_disks):
+    '''
+    Print the current state :
+      - what devices matched
+      - what devices didn't matched
+    '''
+    bdev_matched = []
+    bdev_unmatched = []
+
+    def prepare_device_string(device):
+        device_string = ""
+        if "ceph_prepared" in device:
+            device_string += " (ceph %s)" % device["ceph_prepared"]
+        device_string += " : "
+        for feature in ["vendor", "model", "size", "rotational"]:
+            if feature in device:
+                device_string += "%s:%s " % (feature[0], device[feature])
+
+        return device_string
+
+    logger.info("Matched devices   : %3d", len(matched_devices))
+    for matched_device in sorted(matched_devices):
+        logger.info(" %s : %s%s", matched_device, matched_devices[matched_device]["bdev"], prepare_device_string(matched_devices[matched_device]))
+        bdev_matched.append(matched_devices[matched_device]["bdev"])
+
+    for physical_disk in sorted(physical_disks):
+        pdisk = physical_disks[physical_disk]
+        if pdisk["bdev"] not in bdev_matched:
+            bdev_unmatched.append("%s %s" % (pdisk["bdev"], prepare_device_string(pdisk)))
+
+    logger.info("Unmatched devices : %3d", len(bdev_unmatched))
+    for bdev in sorted(bdev_unmatched):
+        logger.info(" %s", bdev)
+
+
+def setup_logging():
+    '''
+    Preparing the logging system
+    '''
+    hdlr = logging.FileHandler('/var/log/choose_disk.log')
+    formatter = logging.Formatter('%(asctime)s %(levelname)s %(message)s')
+    hdlr.setFormatter(formatter)
+    logger.addHandler(hdlr)
+    logger.setLevel(logging.INFO)
+    logger.info("############")
+    logger.info("# Starting #")
+    logger.info("############")
+
+
+def fatal(message, module):
+    '''
+    Report a fatal error and exit
+    '''
+    logger.error("### FATAL ###")
+    logger.error(message)
+    logger.error("#############")
+    logger.info("#######")
+    logger.info("# End #")
+    logger.info("#######")
+    if module:
+        module.fail_json(msg=message)
+    else:
+        exit(1)
+
+
+def get_var(module, variable):
+    '''
+    Extract the ansible variable from the playbook
+    If variable doesn't exist, let's return None
+    '''
+    if variable in module.params["vars"]:
+        return module.params["vars"][variable]
+    return None
```

Comment:
no need to return `None` here as functions that do not return anything will implicitly have `None` returned to the caller